source/extensions/filters/network/dubbo_proxy/deserializer.h

```diff
@@ -1,23 +1,52 @@
 #pragma once
 
+#include <string>
+#include <unordered_map>
+
 #include "envoy/buffer/buffer.h"
 
+#include "common/common/assert.h"
+#include "common/config/utility.h"
 #include "common/singleton/const_singleton.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace NetworkFilters {
 namespace DubboProxy {
 
+enum class DeserializerType : uint8_t {
+  Hessian = 0,
+  Json = 1,
+
+  // ATTENTION: MAKE SURE THIS REMAINS EQUAL TO THE LAST PROTOCOL TYPE
+  LastDeserializerType = Json,
+};
+
 /**
  * Names of available deserializer implementations.
  */
 class DeserializerNameValues {
 public:
-  // hessian deserializer
-  const std::string Hessian = "hessian";
-  // json deserializer
-  const std::string Json = "json";
+  struct DeserializerTypeHash {
+    template <typename T> std::size_t operator()(T t) const { return static_cast<std::size_t>(t); }
+  };
+
+  typedef std::unordered_map<DeserializerType, std::string, DeserializerTypeHash>
```

Comment:
ditto, Ideally should be through the serialization type to find the corresponding deserializer

Reply:
I have modified it.